lib/card/set_pattern.rb

```diff
@@ -132,4 +138,23 @@ def rule_set_key
     end
   
   end
+  class TypeSet < SetPattern
+    def initialize card
+      super
+      if mod_key = lookup_inherited_key(card)
+        @inherited_key = mod_key
+      end
+    end
+
+    def lookup_inherited_key card
+      default_rule = card.rule_card(:default) and
```

Comment:
I think I should test at the top for whether the card's type has a codename (or equivalent, module_key) because it will never inherit because it has a type mod bound to the codename.